shared/src/vyxal/RegisterHelpers.scala

```diff
@@ -0,0 +1,64 @@
+package vyxal
+
+import vyxal.conversions.{*, given}
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object RegisterHelpers:
+
+	val register = Context().globals.register
+
+	def clear(): Unit = register.clearAndShrink(1)
+
+	def isEmpty: Boolean = register.length == 0
+
+	def length: VNum = VNum(register.length)
+
+	def reverseRegister(): Unit = 
+		val rev = register.reverse
+		for i <- 0 to register.length - 1 do
+			register.update(i, rev(i))
+		
+	/** Apply a monadic function to an item in the register */	
+	def applyFn(fn: VFun, idx: VNum)(using ctx: Context): Unit =
+		if register.last.isInstanceOf[VPhysical]
+		then
+			val c = Interpreter.executeFn(fn, args = Seq(register(idx.toInt)))
+			register.update(idx.toInt, c)
+
+
+	/** Map a function over the entire register */
+	def map(fn: VFun)(using ctx: Context): Unit =
+		for x <- 0 to register.length - 1 do
+			if register(x).isInstanceOf[VPhysical] then applyFn(fn, VNum(x)) 
+			else register(x)
```

Comment:
The doc comment says one thing, but the function body is doing something else entirely. Shouldn't it be something like this?
```suggestion
	/** Map a function over the entire register */
	def map(fn: VFun)(using ctx: Context): Unit =
		register.forEach(x => applyFn(fn, x))
```
Although I would say this is *applying* a function to the entire register, not *mapping* a function over it. I'm assuming it's more useful to not only apply a function with side effects to the register but to actually modify the register. For that, you could use [`register.mapInPlace(applyFn(fn, _))`](https://www.scala-lang.org/api/current/scala/collection/mutable/ArrayBuffer.html#mapInPlace-fffff965)